llvm/lib/CodeGen/SelectionDAG/DAGCombiner.cpp

```diff
@@ -7166,7 +7166,7 @@ SDValue DAGCombiner::visitAND(SDNode *N) {
 
   // if (and x, c) is known to be zero, return 0
   unsigned BitWidth = VT.getScalarSizeInBits();
-  ConstantSDNode *N1C = isConstOrConstSplat(N1);
+  ConstantSDNode *N1C = isConstOrConstSplat(N1, false, true);
```

Comment:
nit: I find it helpful to label `bool` parameters.
```suggestion
  ConstantSDNode *N1C = isConstOrConstSplat(N1, /*AllowUndefs=*/false, /*AllowTruncation=*/true);
```